src/diagnostics.ts

```diff
@@ -4,14 +4,11 @@ import { Diagnostic, DiagnosticSeverity, Range } from 'vscode-languageserver'
 /**
  * Converts a TypeScript Diagnostic to an LSP Diagnostic
  */
-export function convertTsDiagnostic(diagnostic: ts.Diagnostic): Diagnostic {
+export function convertTsDiagnostic(diagnostic: ts.DiagnosticWithLocation): Diagnostic {
     const text = ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n')
-    let range: Range = { start: { character: 0, line: 0 }, end: { character: 0, line: 0 } }
-    if (diagnostic.file && diagnostic.start && diagnostic.length) {
```

Comment:
Here you first set the initial lines for start and end to zero. 
Then you check if file and start and **length** are not _null/undefined/**0**_. 
In case the file has a syntax error, length sent from TS server is zero. So the value for range doesn't get recalculated. 
Something like
```
class Test {

}

const test = new Test();

test.
```
The last line should give you the syntax error line **Identifier expected**, but the range will be invalid - it'll start at line 0 instead of line 7.